### PR TITLE
[Prompty] Format prompty output

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
Format prompty output:
- Directly return llm response
```
---
name: Basic Prompt

model: 
  api: chat
  parameters:
    max_tokens: 45
  response: all
 
inputs:
  firstName: Sally
  lastName: Doe 

---
system:
You are an AI assistant who helps people find information.
user:
{{question}}
```

- Return json obj when repsonse_format is json_object
```
---
name: Basic Prompt

model: 
  api: chat
  parameters:
    max_tokens: 45
    response_format:
        type: json_object
 
inputs:
  firstName: Sally
  lastName: Doe 
---
system:
As the assistant, you answer questions briefly, succinctly. Your structured response. Only accepts JSON format, likes below:
{"name": customer_name, "answer": the answer content}
user:
{{question}}
```
Prompty output:
{"name": "Sally", "answer": 2}

- Return specified output in json object
```
---
name: Basic Prompt

model: 
  api: chat
  parameters:
    max_tokens: 45
    response_format:
        type: json_object
 
inputs:
  firstName: Sally
  lastName: Doe 

outputs:
    answer:
        type: string
---
system:
As the assistant, you answer questions briefly, succinctly. Your structured response. Only accepts JSON format, likes below:
{"name": customer_name, "answer": the answer content}
user:
{{question}}
```
Prompty output:
{"answer": 2}

- Stream output
```
---
name: Basic Prompt

model: 
  api: chat
  parameters:
    max_tokens: 45
    stream: true
 
inputs:
  firstName: Sally
  lastName: Doe 
---
system:
You are an AI assistant who helps people find information.
user:
{{question}}
```

```
result = prompty(question="what is the result of 1+1?")
result_content = ""
for item in result:
    if len(item.choices) > 0 and item.choices[0].delta.content:
           result_content += item.choices[0].delta.content
```

Prompty trace:
image

convert_prompt_template trace
image
send_request_to_llm trace:
image

image

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/core/_prompty_utils.py [hunk 1]
```diff
@@ -123,21 +124,51 @@ def send_request_to_llm(client, api, parameters):
...
-    if raw or streaming:
+def format_llm_response(response, api, is_first_choice, response_format=None, streaming=False, outputs=None):
+    """
+    Format LLM response
+
+    :param response: LLM response.
+    :type response:
+    :param api: API type of the LLM.
+    :type api: str
+    :param is_first_choice: If true, it will return the first item in response choices, else it will return ll response
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/_prompty_utils.py:135]: "return ll response", what's ll?

File: src/promptflow-core/promptflow/executor/_prompty_executor.py [hunk 2]
```diff
@@ -34,7 +35,7 @@ def _initialize_function(self):
         as executor input.
         """
         # If the function is not decorated with trace, add trace for it.
-        func = _traced(self.prompty)
+        func = _traced(self.prompty, trace_type=TraceType.FLOW)
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/executor/_prompty_executor.py:38]: shall we add a new type for Promty? Is there any thing we can tell this is a prompty in trace UI?

File: src/promptflow-core/promptflow/core/_prompty_utils.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/core/_prompty_utils.py]: fix typo

File: src/promptflow-core/promptflow/executor/_prompty_executor.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/executor/_prompty_executor.py]: Add a new type for prompty.
image

File: src/promptflow-tracing/promptflow/tracing/_trace.py [hunk 3]
```diff
@@ -282,7 +282,7 @@ def serialize_attribute(value):
 
 
 def _traced(
-    func: Callable = None, *, args_to_ignore: Optional[List[str]] = None, trace_type=TraceType.FUNCTION
+    func: Callable = None, *, args_to_ignore: Optional[List[str]] = None, trace_type=TraceType.FUNCTION, name=None
```
[the remark this answers is not in the document]
Comment (lalala123123) [reply src/promptflow-tracing/promptflow/tracing/_trace.py:285]: add some screenshots to description. After completed this PR, I will check if these trace information need to be optimized in a new PR

**Review by 0mza987 — DISMISSED** [review]

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — DISMISSED** [review]

**Review by 0mza987 — DISMISSED** [review]

**Review by 0mza987 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**